providers/apollo/metadata.go

```diff
@@ -0,0 +1,90 @@
+package apollo
+
+import (
+	"context"
+	"errors"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+// ListObjectMetadata creates metadata of object via reading objects using Apollo API.
+func (c *Connector) ListObjectMetadata(ctx context.Context,
+	objectNames []string,
+) (*common.ListObjectMetadataResult, error) {
+	// Ensure that objectNames is not empty
+	if len(objectNames) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	metadataResult := common.ListObjectMetadataResult{
+		Result: make(map[string]common.ObjectMetadata),
+		Errors: make(map[string]error),
+	}
+
+	for _, obj := range objectNames {
+		url, err := c.getAPIURL(obj)
+		if err != nil {
+			return nil, err
+		}
+
+		resp, err := c.Client.Get(ctx, url.String())
+		if err != nil {
+			metadataResult.Errors[obj] = err
+
+			continue
+		}
+
+		// Check nil response body, to avoid panic.
+		if resp == nil || resp.Body == nil {
+			metadataResult.Errors[obj] = common.ErrEmptyResponse
+
+			continue
+		}
+
+		metadata, err := parseMetadataFromResponse(resp, obj)
+		if err != nil {
+			if errors.Is(err, common.ErrMetadataLoadFailure) {
+				metadataResult.Errors[obj] = common.ErrEmptyResponse
+			} else {
+				return nil, err
+			}
+		}
+
+		metadata.DisplayName = obj
+		metadataResult.Result[obj] = metadata
+	}
+
+	return &metadataResult, nil
+}
+
+func parseMetadataFromResponse(resp *common.JSONHTTPResponse, obj string) (common.ObjectMetadata, error) {
+	bb := resp.Body.Source()
+	if bb == nil {
+		return common.ObjectMetadata{}, common.ErrEmptyResponse
+	}
+
+	metadata := common.ObjectMetadata{
+		FieldsMap: make(map[string]string),
+	}
+
+	root, err := ajson.Unmarshal(bb)
+	if err != nil {
+		return common.ObjectMetadata{}, err
+	}
+
+	resultArr, err := jsonquery.New(root).Array(obj, true)
+	if err != nil {
+		return common.ObjectMetadata{}, err
+	}
+
+	objectResponnse := resultArr[0].MustObject()
```

Comment:
Are there any rate limits / quota considerations we need to take into account? Because this will count against it. We could very well just have a static metadata file like we have for some other connectors

Reply:
Yes, there is (depending on plan) the minimum(free trial) is 50 calls/min, 200/hour and 600/day. So this will have an effect on it. Was looking into static metadata file as well. But couldn't find  specs files. If you agree, Let me create a script that does this for all objects and create the static file for it. So that we wouldn't be doing this for every integration.